### DISCUSSION TITLE
Question about turbinflow in pelec

### DISCUSSION THREAD

**Discussion by Svalencia303** [AMReX-Combustion/PeleC#656] [Q&A] [answered]
In PeleC, I am facing difficulties while attempting to generate a turbulent field using the turbinflow tool. My initial intention was to simulate the JetFlame case, but it prompted me to provide an additional file. For this, I used the same file that was created for the PeleLMeX/Exec/RegTests/TurbInflow/TurbFileHIT/ case. However, I am uncertain if this is the correct approach.

Interestingly, when I conducted a simulation in PeleLMeX, I could observe the presence of turbulence. However, in PeleC, the turbulence does not seem to manifest. I suspect that the turbulence file generated in PeleLMeX might not be suitable for PeleC.

To clarify the process and ensure I am using turbinflow correctly in PeleC, I would appreciate some guidance on the proper procedure.

**Comment by marchdf** [comment]
Could this be due to a units thing? PeleC is cgs units. You can see how we do turb inflow in PeleC here: https://github.com/AMReX-Combustion/PeleC/tree/development/Exec/Production/ChallengeProblem and you will notice some scaling MKS->cgs here: https://github.com/AMReX-Combustion/PeleC/blob/development/Exec/Production/ChallengeProblem/challenge.inp#L149

**Comment by drummerdoc** [comment]
Note that you may need to scale both the spatial coordinate used to interpolate the file data, and the velocity that results.

**Reply by marchdf** [reply]
yes, to this point, you need to modify the `Turb.test2/HDR` file and the spatial coordinate on L2.

**Comment by Svalencia303** [comment]
the settings I used are as follows: 

turbinflows = bottom
turbinflow.bottom.turb_file = TurbFileHIT/TurbTEST
turbinflow.bottom.dir = 2
turbinflow.bottom.side = low
turbinflow.bottom.turb_scale_loc = 100
turbinflow.bottom.turb_scale_vel = 100
turbinflow.bottom.turb_center = 0.0 0.0 
turbinflow.bottom.turb_conv_vel = 50
turbinflow.bottom.turb_nplane = 32

do I need to do a scaling somewhere else ?  

additionally I would like to know what does this last parameter turbinflow.bottom.turb_nplane refers to?

**Reply by marchdf** [reply]
that's doesn't really match the settings I pointed to in my answer above. Our input file does not change `scale_loc` and your `turb_conv_vel` looks very low. Can you maybe start with the problem I pointed to and then modify it to make it similar to the case you are trying to run?

**Comment by drummerdoc** [comment]
nplanes: The data file could be arbitrarily large which would consume an unnecessarily large amount of memory.  So, we added the ability to only read in part of the file at a time.  The reader/interpolater manages ensuring that the needed planes are in memory.  If not, it dumps and reads a set that does.

**Comment by baperry2** [comment]
We do need to improve the documentation on this. I opened an issue in PelePhysics (where the turbinflow capability is implemented) to add some documentation on this capability: https://github.com/AMReX-Combustion/PelePhysics/issues/401

**Comment by Svalencia303** [comment]
I have been trying to simulate the jet flame and challenge problem cases, however in neither case can I get turbulence. I am thinking that the problem may be the HDR and DAT files I got from pelelmex/turbinflow, any advice on how I can change those files and adjust them for the simulated cases?

**Reply by marchdf** [reply]
my HDR file has this at the top:
```
131 131 1025
0.0203125 0.0203125 0.16
1 1 1
0
137378
```

**Reply by baperry2** [reply, marked as answer]
I looked into the JetFlame case a bit and was able to get it to push in turbulence based on the files generated from turbinflow in PeleLMeX. That generates turbulence in a box of size 2pi*2pi*2pi and a u_rms of 1.0 (both arbitrary units). The HDR file starts like:

> $ cat Turb.test2/HDR
131 131 129
6.33151 6.33151 6.2341
1 1 1
0
137378
...

The jet diameter in the JetFlame case is `prob.D = 1.7e-2` (cm). So to get the turbulence patch to be the same size as the jet, we need to rescale by 2pi/1.7e-2 = 369.5 (set `turbinflow.bottom_jet.turb_scale_loc = 369.5`). The jet velocity is set through *both* `prob.v_in = 2800` and `turbinflow.bottom_jet.turb_conv_vel  = 2800.0` in units of (cm/s). To get velocity fluctuations that show up relative to that, u_rms should be say 280 cm/s (for a turbulence intensity of 10%). That means we need to set `turbinflow.bottom_jet.turb_scale_vel = 280.0`. Also set `turbinflow.bottom_jet.turb_nplane    = 64` so we don't request to load more planes at once than are in the turbulence file. If you adjust those parameters and run for ~50 timesteps you should see velocity fluctuations coming in (easiest to see in the z-velocity). After confirming that, you can update the jet velocity and turb_scale_vel to match your conditions of interest.

**Reply by drummerdoc** [reply]
thanks for stepping through that Bruce!

**Comment by Svalencia303** [comment]
thank you very much for your help, I was able to replicate the cases.
